File: 1695-maximum-erasure-value/1695-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        s = set()
        l, r, output, currSum = 0, 0, 0, 0
        while r < len(nums):
            while nums[r] in s:
                s.remove(nums[l])
                currSum -= nums[l]
                l += 1
            s.add(nums[r])
            currSum += nums[r]
            output = max(output, currSum)
            r += 1
        return output

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        bool arr[10001] = { false };
        int l = 0, r = 0, output = 0, currSum = 0;
        while (r < nums.size()) {
            while (arr[nums[r]]) {
                arr[nums[l]] = false;
                currSum -= nums[l++];
            }
            arr[nums[r]] = true;
            currSum += nums[r++];
            output = max(output, currSum);
        }
        return output;
    }
};